src/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs

```diff
@@ -34,8 +35,13 @@ public void Ctor_NullSamAccountName_ThrowsArgumentException()
         [Fact]
         public void Ctor_EmptySamAccountName_ThrowsArgumentNullException()
         {
-            var context = new PrincipalContext(ContextType.Machine);
-            AssertExtensions.Throws<ArgumentNullException>("Principal.SamAccountName cannot be null or empty.", () => new ComputerPrincipal(context, string.Empty, "password", enabled: true));
+            RemoteInvoke(() =>
+            {
+                CultureInfo.CurrentUICulture = CultureInfo.InvariantCulture;
+
+                var context = new PrincipalContext(ContextType.Machine);
+                AssertExtensions.Throws<ArgumentNullException>("Principal.SamAccountName cannot be null or empty.", () => new ComputerPrincipal(context, string.Empty, "password", enabled: true));
```

Comment:
It is a bug (in both NETFx and Core) that it is providing a message for the parameter name. So an alternative might be to not validate the parameter name, or validate it only in Core and change the code in Core to separate parameter name and message.